Pertanyaan: Kasih komen penejelasan di code pertemuan 6, interpretasikan hasil R2 Score dan MSE nya (ezz)

# Import Libraries

In [2]:
from sklearn.datasets import load_diabetes
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, r2_score

Mengimpor library sklearn untuk melakukan regresi linear pada dataset diabetes.
- 'load_diabetes' memanggil dataset diabetes.
- 'LinearRegression' melakukan regresi linear pada dataset diabetes.
- 'train_test_split' mengimpor fungsi untuk membagi dataset menjadi data pelatihan dan pengujian.
- 'mean_squared_error dan r2_score' mengimpor fungsi untuk mengevaluasi performa model.

In [3]:
x, y = load_diabetes(return_X_y=True)

Membagi dataset diabetes menjadi fitur (X) dan target (y)

# Linear Regression

Melakukan analisis data diabetes menggunakan model regresi linear

In [4]:
# Membuat Variabel dengan nilai berupa model regresi linear
lr = LinearRegression()

In [6]:
# Melatih model regresi linear menggunakan data fitur (x) dan target (y).
lr.fit(x,y)
y_pred = lr.predict(x)

In [7]:
# Mengakses nama-nama fitur
load_diabetes()['feature_names']

['age', 'sex', 'bmi', 'bp', 's1', 's2', 's3', 's4', 's5', 's6']

In [17]:
# Mengakses nilai target
load_diabetes()['target']

array([151.,  75., 141., 206., 135.,  97., 138.,  63., 110., 310., 101.,
        69., 179., 185., 118., 171., 166., 144.,  97., 168.,  68.,  49.,
        68., 245., 184., 202., 137.,  85., 131., 283., 129.,  59., 341.,
        87.,  65., 102., 265., 276., 252.,  90., 100.,  55.,  61.,  92.,
       259.,  53., 190., 142.,  75., 142., 155., 225.,  59., 104., 182.,
       128.,  52.,  37., 170., 170.,  61., 144.,  52., 128.,  71., 163.,
       150.,  97., 160., 178.,  48., 270., 202., 111.,  85.,  42., 170.,
       200., 252., 113., 143.,  51.,  52., 210.,  65., 141.,  55., 134.,
        42., 111.,  98., 164.,  48.,  96.,  90., 162., 150., 279.,  92.,
        83., 128., 102., 302., 198.,  95.,  53., 134., 144., 232.,  81.,
       104.,  59., 246., 297., 258., 229., 275., 281., 179., 200., 200.,
       173., 180.,  84., 121., 161.,  99., 109., 115., 268., 274., 158.,
       107.,  83., 103., 272.,  85., 280., 336., 281., 118., 317., 235.,
        60., 174., 259., 178., 128.,  96., 126., 28

In [18]:
# Menampilkan nilai R-Squared
print(r2_score(y,y_pred))

0.5177484222203499


In [19]:
# Menampilkan nilai MSE
print(mean_squared_error(y, y_pred))

2859.69634758675


# Data Splitting

In [21]:
# Membagi dataset menjadi dua bagian: 80% data latih dan 20% data uji.  
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42)

In [22]:
# Melatih model regresi linear dan mengevaluasi kinerjanya menggunakan metrik R² dan Mean Squared Error (MSE).
lr = LinearRegression()
lr.fit(X_train, y_train)
y_pred = lr.predict(X_test)
print(r2_score(y_test,y_pred))
print(mean_squared_error(y_test, y_pred))

0.4526027629719197
2900.19362849348


Interpretasi: Nilai R² berkisar antara 0 hingga 1 dengan ketentuan semakin mendekati angka satu berarti semakin baik. Dalam kasus ini, R² sebesar 0.4526 menunjukkan bahwa sekitar 45.26% dari variasi penyakit diabetes dapat dijelaskan oleh model regresi linear. Sementara itu, Mean Squared Error (MSE) mengukur rata-rata dari kuadrat selisih antara nilai yang diprediksi dan nilai yang sebenarnya. MSE yang lebih rendah menunjukkan model yang lebih baik. Nilai MSE sebesar 2900.19 prediksi model menyimpang sekitar 53.85 (akar dari MSE) dari nilai aktual. Saran yang bisa dilakukan untuk mempertimbangkan beberapa langkah untuk meningkatkan kinerja model adalah dengan mencoba algoritma regresi lain, misalnya regresi ridge, lasso, atau decision tree.